scripts/verify.py

```diff
@@ -6,32 +6,38 @@
 import argparse
 
 from subprocess import run
+from subprocess import CalledProcessError
 import os
 import shutil
 
 def main(checks):
-    print("Verifying with " + str(checks))
-    if "pytest" in checks:
-        print("Tests (pytest):", flush=True)
-        run("pytest -v --color=yes", shell=True, check=True)
-
-    if "pylint" in checks:
-        print("Linter (pylint):", flush=True)
-        run("pylint -d locally-disabled,locally-enabled -f colorized allennlp tests", shell=True, check=True)
-        print("pylint checks passed")
-
-    if "mypy" in checks:
-        print("Typechecker (mypy):", flush=True)
-        run("mypy allennlp --ignore-missing-imports", shell=True, check=True)
-        print("mypy checks passed")
-
-    if "build-docs" in checks:
-        print("Documentation (build):", flush=True)
-        run("cd docs; make html-strict", shell=True, check=True)
-
-    if "checks-docs" in checks:
-        print("Documentation (check):", flush=True)
-        run("./script/check_docs.py", shell=True, check=True)
+    try:
+        print("Verifying with " + str(checks))
+        if "pytest" in checks:
+            print("Tests (pytest):", flush=True)
+            run("pytest -v --color=yes", shell=True, check=True)
+
+        if "pylint" in checks:
+            print("Linter (pylint):", flush=True)
+            run("pylint -d locally-disabled,locally-enabled -f colorized allennlp tests", shell=True, check=True)
+            print("pylint checks passed")
+
+        if "mypy" in checks:
+            print("Typechecker (mypy):", flush=True)
+            run("mypy allennlp --ignore-missing-imports", shell=True, check=True)
+            print("mypy checks passed")
+
+        if "build-docs" in checks:
+            print("Documentation (build):", flush=True)
+            run("cd docs; make html-strict", shell=True, check=True)
+
+        if "checks-docs" in checks:
+            print("Documentation (check):", flush=True)
+            run("./script/check_docs.py", shell=True, check=True)
+
+    except CalledProcessError:
+        # squelch
+        pass
```

Comment:
If you `pass` here, this script won't work for CI, because there's no indication of failure.  You need to do `sys.exit(-1)`.  Also, having the `try` block on the outside will stop the checks at the first failure; seems reasonable to want to run all of them even if one fails.

Reply:
I'll try to return the exit code of the subprocess.

@matt-gardner I had intended for this script to exit when one command fails, but if you'd prefer I can make it run all of them.  I wasn't planning on running this directly with `--all` from TeamCity.

Reply:
You're the one that's going to be using it, so as long as TeamCity still runs all of the checks even if one fails, it doesn't much matter to me if this script exits when one fails - whatever works for you.